packages/flipper-plugin-react-native-performance/android/FlipperReactPerformancePlugin.java

```diff
@@ -0,0 +1,154 @@
+    }
+
+    private void measureNativeStartupTime() {
+        try {
+            long processStartTime = FlipperReactPerformancePlugin.getStartTime(Process.myPid());
+            nativeStartupDuration = SystemClock.elapsedRealtime() - processStartTime;
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void addReactMarkers() {
+        ReactMarker.addListener(
+                (name, tag, instanceKey) -> {
+                    switch (name) {
+                        case DOWNLOAD_START:
+                            scriptDownloadStart = System.currentTimeMillis();
+                            break;
+                        case DOWNLOAD_END:
+                            scriptDownloadEnd = System.currentTimeMillis();
+                            sendMeasurements();
+                            break;
+                        case RUN_JS_BUNDLE_START:
+                            scriptExecutionStart = System.currentTimeMillis();
+                            break;
+                        case RUN_JS_BUNDLE_END:
+                            scriptExecutionEnd = System.currentTimeMillis();
+                            measureBundleSize();
+                            sendMeasurements();
+                            break;
+                        case CONTENT_APPEARED:
+                            contentAppeared = System.currentTimeMillis();
+                            sendMeasurements();
+                            break;
+                        case RELOAD:
+                            sessionStartTime = System.currentTimeMillis();
+                            scriptDownloadStart = 0;
+                            scriptDownloadEnd = 0;
+                            scriptExecutionStart = 0;
+                            scriptExecutionEnd = 0;
+                            contentAppeared = 0;
+                            sendMeasurements();
+                            break;
+                    }
+                });
+    }
+
+    private static long getStartTime(final int pid) throws IOException {
+        final String path = "/proc/" + pid + "/stat";
+        final BufferedReader reader = new BufferedReader(new FileReader(path));
+        final String stat;
+        try {
+            stat = reader.readLine();
+        } finally {
+            reader.close();
+        }
+        final String field2End = ") ";
+        final String fieldSep = " ";
+        final int fieldStartTime = 20;
+        final int msInSec = 1000;
+        try {
+            final String[] fields = stat.substring(stat.lastIndexOf(field2End)).split(fieldSep);
+            final long t = Long.parseLong(fields[fieldStartTime]);
+            final long tck;
+            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
+                tck = Os.sysconf(OsConstants._SC_CLK_TCK);
+            } else {
+                tck = 0;
+            }
+            return (t * msInSec) / tck;
```

Comment:
Instead of setting `tck` to `0` and causing a division by zero, I would suggest to throw directly in a guard clause, or finding a different logic in the else-branch.

That would make the stack trace a bit clearer, skipping the `ArithmeticException`

Reply:
removed.